app/scripts/main.js

```diff
@@ -392,6 +402,84 @@ var iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
       }
       return def.promise();
     },
+    getScatterData: function(_self, groupId) {
+      var def = new $.Deferred();
+      var self = this;
+      var data = self.getGroupNdx(groupId);
+
+      $.when(window.iviz.datamanager.getStudyToSampleToPatientdMap()).then(function(_studyToSampleToPatientMap) {
+        $.when(window.iviz.datamanager.getCaseLists()).then(function(_caseLists) {
+          var _sequencedCaseUIDs = [];
+          var _allCaseUIDs = [];
+          var _sequencedCaseUIdsMap = {};
+          
+          $.each(_caseLists, function(studyId, caseList) {
+            if (caseList.sequencedSampleIds.length > 0) {
+              $.each(caseList.sequencedSampleIds, function(index, sampleId) {
+                _sequencedCaseUIDs.push(_studyToSampleToPatientMap[studyId].sample_to_uid[sampleId]);
+              });
+            }
+            if (caseList.allSampleIds.length > 0) {
+              $.each(caseList.allSampleIds, function(index, sampleId) {
+                _allCaseUIDs.push(_studyToSampleToPatientMap[studyId].sample_to_uid[sampleId]);
+              });
+            }
+          });
+          _sequencedCaseUIDs = _sequencedCaseUIDs.length > 0 ? _sequencedCaseUIDs : _allCaseUIDs;
+          _.each(_sequencedCaseUIDs, function(_sampleUid) {
+            _sequencedCaseUIdsMap[_sampleUid] = _sampleUid;
+          });
+
+          $.when(window.iviz.datamanager.getCnaFractionData(),
+            window.iviz.datamanager.getMutationCount())
+            .then(function(_cnaFractionData, _mutationCountData) {
+              var _hasCNAFractionData = _.keys(_cnaFractionData).length > 0;
+              var _hasMutationCountData = _.keys(_mutationCountData).length > 0;
+
+              if ((!_hasCNAFractionData || !_hasMutationCountData) && firstScatterShow) {
```

Comment:
firstScatterShow is a flag. When the page is loaded, firstScatterShow is true so the code will remove empty chart and it will be changed to false. When a user wants to add an empty chart, it won't run into removing empty chart part since firstScatterShow is false even if (!_hasCNAFractionData || !_hasMutationCountData) is always true.